### PR TITLE
Add C implementation for `@stdlib/math/base/special/atan`

### LINKED ISSUE #696: [RFC]: Add C implementation for `@stdlib/math/base/special/atan`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/atan`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/atan)
```c
double stdlib_base_atan( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #696 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All required checks cleared, this PR can have a final review :)

File: lib/node_modules/@stdlib/math/base/special/atan/README.md [hunk 1]
```diff
@@ -74,6 +74,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+```
+
+#### stdlib_base_atan( x )
+
+Computes the [arctangent][arctangent] of a `number`.
+
+```c
+double out = stdlib_base_atan( 0.0 );
+// returns ~0.0
+out = stdlib_base_atan( -3.141592653589793/2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/README.md:110]:
```suggestion

out = stdlib_base_atan( -3.141592653589793/2.0 );
```

File: lib/node_modules/@stdlib/math/base/special/atan/README.md [hunk 2]
```diff
@@ -74,6 +74,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/atan.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/README.md:144]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/atan/README.md [hunk 3]
```diff
@@ -74,6 +74,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+    double x[] = { -1000, -777.78, -555.56, -333.33, -111.11,  111.11, 333.33,  555.56, 777.78, 1000 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/README.md:145]:
```suggestion
    double x[] = { -1000.0, -777.78, -555.56, -333.33, -111.11,  111.11, 333.33,  555.56, 777.78, 1000.0 };
```

File: lib/node_modules/@stdlib/math/base/special/atan/benchmark/c/native/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+int main( void ) {
+	double elapsed;
+	int i;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/benchmark/c/native/benchmark.c:130]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/atan/examples/c/example.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/atan.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { -1000, -777.78, -555.56, -333.33, -111.11,  111.11, 333.33,  555.56, 777.78, 1000 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/examples/c/example.c:24]:
```suggestion
    double x[] = { -1000.0, -777.78, -555.56, -333.33, -111.11,  111.11, 333.33,  555.56, 777.78, 1000.0 };
```

File: lib/node_modules/@stdlib/math/base/special/atan/src/main.c [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+static double polyval_q( const double x ) {
+	return 194.5506571482614 + (x * (485.3903996359137 + (x * (432.88106049129027 + (x * (165.02700983169885 + (x * (24.858464901423062 + (x * 1.0)))))))));
+}
+
+// END: polyval_q
+
+/* End auto-generated functions. */
+
+/**
+* Evaluates the natural logarithm of \\(1+x\\).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/src/main.c:89]:
```suggestion
* Computes the arctangent of a number.
```

File: lib/node_modules/@stdlib/math/base/special/atan/src/main.c [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+*
+* @param x    input value
+* @return	  output value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atan/src/main.c:92]:
```suggestion
* @return	  arctangent (in radians)
```

**Review by kgryte — APPROVED** [review]
LGTM. As the requested changes are minor and non-blocking, I'll merge and touch up on my end. Thanks, @Pranavchiku!

**Pull request merged**